### PR TITLE
Productionising checklist complete

### PR DESCRIPTION
**Pull request by humulla**
A couple of changes was needed in order for API gateway test to work in Production:

- [x] Canaries has been set up
- [x] API keys has been added
- [x] Create env var for API keys

### CODE REVIEW

File: ResidentInformationApi/ResidentInformationApi.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
         <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
         <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
         <RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
-        <AssemblyName>ResidentVulnerabilitiesApi</AssemblyName>
+        <AssemblyName>ResidentInformationApi</AssemblyName>
```
Comment (Renny4Real) [inline ResidentInformationApi/ResidentInformationApi.csproj:12]: Nice Spot!

File: ResidentInformationApi/V1/Gateways/Helpers/DictionaryBuilder.cs [hunk 2]
```diff
@@ -10,10 +10,10 @@ public static string BuildQueryDictionary(ResidentQueryParam rqp)
         {
             var queryDictionary = new Dictionary<string, string>();
 
-            if (!string.IsNullOrEmpty(rqp.FirstName)) queryDictionary.Add("first_name", rqp.FirstName);
-            if (!string.IsNullOrEmpty(rqp.LastName)) queryDictionary.Add("last_name", rqp.LastName);
-            if (!string.IsNullOrEmpty(rqp.Address)) queryDictionary.Add("address", rqp.Address);
-            if (!string.IsNullOrEmpty(rqp.Postcode)) queryDictionary.Add("postcode", rqp.Postcode);
+            if (!string.IsNullOrWhiteSpace(rqp.FirstName)) queryDictionary.Add("first_name", rqp.FirstName);
```
Comment (Renny4Real) [inline ResidentInformationApi/V1/Gateways/Helpers/DictionaryBuilder.cs]: Why are we changing it from `IsNullOrEmpty` to `IsNullOrWhiteSpace`?

**Review by elcorbs — COMMENTED** [review]
Looks good, Can you also add tests to test that the API KEY is being used when making requests to the API's?

File: ResidentInformationApi/serverless.yml [hunk 3]
```diff
@@ -20,12 +20,17 @@ package:
 functions:
   residentInformationApi:
     name: ${self:service}-${self:provider.stage}
+    timeout: 15
```
Comment (Renny4Real) [inline ResidentInformationApi/serverless.yml:23]: Is this a lambda timeout for 15s? Why 15?

File: terraform/staging/main.tf [hunk 4]
```diff
@@ -7,10 +7,6 @@ provider "aws" {
 }
 data "aws_caller_identity" "current" {}
 data "aws_region" "current" {}
-locals {
```
Comment (Renny4Real) [inline terraform/staging/main.tf:10]: Why are we removing the `locals`?

**Review by Renny4Real — COMMENTED** [review]
Looks good 🎉

File: ResidentInformationApi/V1/Gateways/Helpers/DictionaryBuilder.cs (hunk 2 above)
Comment (humulla) [reply ResidentInformationApi/V1/Gateways/Helpers/DictionaryBuilder.cs]: Good spot, this was changed for some testing, forgot to change it back, will do so now

File: ResidentInformationApi/serverless.yml (hunk 3 above)
Comment (humulla) [reply ResidentInformationApi/serverless.yml:23]: Yeh, unfortunately due to API going through to different APIs it takes longer than 6s, which makes sense so had to be extended

File: terraform/staging/main.tf (hunk 4 above)
Comment (humulla) [reply terraform/staging/main.tf:10]: I'm not too sure, I think louise removed it, but not for the reason, don't see a documentation on it too

File: ResidentInformationApi/serverless.yml (hunk 3 above)
Comment (Renny4Real) [reply ResidentInformationApi/serverless.yml:23]: Cool. 👍 
Yeah, that makes sense. Might be something to watch, especially if we start adding more APIs.

File: terraform/staging/main.tf (hunk 4 above)
Comment (Renny4Real) [reply terraform/staging/main.tf:10]: Cool. 👍 There might be a reason it was removed. Not sure either.

**Review by LBHBomotoso — DISMISSED** [review]

File: ResidentInformationApi.Tests/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (LBHBomotoso) [inline ResidentInformationApi.Tests/Properties/launchSettings.json:1]: I think this can be deleted?

**Review by elcorbs — APPROVED** [review]

File: ResidentInformationApi.Tests/Properties/launchSettings.json (hunk 5 above)
Comment (humulla) [reply ResidentInformationApi.Tests/Properties/launchSettings.json:1]: The whole file?

Comment (LBHBomotoso) [reply ResidentInformationApi.Tests/Properties/launchSettings.json:1]: I'm not sure, is it used?

Comment (humulla) [reply ResidentInformationApi.Tests/Properties/launchSettings.json:1]: Not sure, but it looks like it's not as it's got transaction API init?

Comment (LBHBomotoso) [reply ResidentInformationApi.Tests/Properties/launchSettings.json:1]: It looks like the other APIs have it. I guess it's fine.

Comment (humulla) [reply ResidentInformationApi.Tests/Properties/launchSettings.json:1]: I've just tested it without it and it works, will remove it as i can be confusing

Comment (humulla) [reply ResidentInformationApi.Tests/Properties/launchSettings.json:1]: Oh okay, I'll leave it then

**Pull request merged**